Code/LauncherUnified/Launcher.cpp

```diff
@@ -251,6 +251,9 @@ namespace O3DELauncher
                 }
             };
             AZ::Data::AssetCatalogRequestBus::Broadcast(AZStd::move(LoadCatalog));
+
+            // Broadcast that critical assets are ready
+            AZ::ComponentApplicationLifecycle::SignalEvent(*settingsRegistry, "CriticalAssetsCompiled", R"({})");
```

Comment:
This implies that all critical assets are completely ready the moment you load the catalog.  is this actually true?  What if you start without a cache and start AP, then immediately start the app via VFS?  It could be a while after connection but before critical assets are actually compiled... or does this part of the code only execute after some other part of the code has already verified that all critical assets are done?

Reply:
The [ConnectToAssetProcessor](https://github.com/o3de/o3de/blob/development/Code/LauncherUnified/Launcher.cpp#L273-L274) function below,  connects to the AssetProcessor using the `AzFramework::AssetSystem::ConnectionSettings` struct.
By default that waits until the [AP is ready](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Asset/AssetSystemBus.h#L116-L118), unless that option has been overridden in the [bootstrap.setreg](https://github.com/o3de/o3de/blob/development/Registry/bootstrap.setreg) file.

So it is left up to the Project Maintainer to determine if they do not want to wait for the AP to enter the Ready state after compiling Critical Assets